docs/blueprints/storage-node-downtime-disqualification.md

```diff
@@ -0,0 +1,112 @@
+
+This implementation still exhibits the problem that, for nodes whose last contact was over one hour ago, we could concievably give them a full punishment even if they were only offline for one minute.
+
+## Implementation
+
+1. Remove old uptime reputation values from codebase:
+
+    uptime_success_count<br>
+    total_uptime_count<br>
+    uptime_reputation_alpha<br> 
+    uptime_reputation_beta<br>
+
+2. Add `downtime_suspended` and `under_review` timestamp columns to the nodes table and rename the `suspended` column to `audit_suspended`
+
+3. Implement new logic in estimation chore to suspend nodes
+    - Refactor downtimeTrackingDB method `GetOfflineTime` to check for entries which contain downtime from outside the measurement period
+
+        There is a particular characteristic of downtime tracking to take note of here:<br>
+        
+        A downtime entry tracked at March 2nd 00:00 contains downtime for some length of time preceding that point. See [downtime tracking](storage-node-downtime-tracking.md)<br>
+        
+        If we want to measure total downtime starting at March 1st 00:00, and there is a March 1st 01:00 entry indicating 2 hours of downtime, we need to truncate the measured downtime in our calculation in order to avoid unfairly including downtime which occurred outside the tracking period.<br>
+        
+        We should be able to solve this problem by taking the first entry within the tracking period and checking if the offline time it contains extends beyond the start of the tracking period. If so, truncate it to the start of the tracking period, then sum the rest of the entries as usual.
+
+    - Refactor relevant overlay methods to handle downtime_suspended nodes: `KnownReliable`, `FindNodes`, etc.
+    - Add notification to SN dashboard indicating suspension for downtime.
+
+4. Implement email notifications if node becomes suspended
+
+5. Implement evaluation chore to reinstate and disqualify nodes
+    - Check total downtime for each suspended node where last_contact_success > last_contact_failure. If the total downtime has fallen below the allowed downtime, the node's suspension is lifted, though it still remains under review.
+    - If node.under_review <= now - (grace period + tracking period) it is eligible for evaluation
+    - If the node is eligible for evaluation, we must also ensure that the entire tracking period is accounted for. If last contact was a failure and occurred before the end of the tracking period, we might have a window of downtime which has not yet been recorded. Thus, to measure all downtime for the tracking period, we must also wait until last_contact_success > failure OR last_contact_failure >= end of tracking period.
```

Comment:
when checking if a node is ready to not be `under_review` and we find a last contact failure, don't we also need to find the node's new total offline time and make sure it doesn't exceed Allowed Downtime? is that what this paragraph is describing? also, do we sum this new window of downtime with the original amount of downtime that got them suspended?